homework/index.js

```diff
@@ -30,14 +30,120 @@
+    }
+  }
+
+  function createInfoList(repository, container) {
+    clearContainer(container);
+    const div = createAndAppend('div', container, { id: 'info_div' });
+    const table = createAndAppend('table', div);
+    const tbody = createAndAppend('tbody', table);
+    createAndAppend('tr', table);
+    createAndAppend('tr', table, {});
+    addRow(tbody, 'Repository:', repository);
+    addRow(tbody, 'Description:', repository);
+    addRow(tbody, 'Forks:', repository);
+    addRow(tbody, 'Updated:', repository);
+  }
+
+  function createContributorsList(repositoryName, ul) {
+    clearContainer(ul);
+    const url = `https://api.github.com/repos/HackYourFuture/${repositoryName}/contributors`;
+    fetchJSON(url, (err, data) => {
+      if (err) {
+        renderError(err);
+        return;
+      }
+
+      createAndAppend('h2', ul, {
+        text: 'contributions',
+      });
+      data.map(item => {
+        const li = createAndAppend('li', ul);
+        const div = createAndAppend('div', li);
+
+        createAndAppend('img', div, {
+          class: 'contributor_avatar',
+          src: item.avatar_url,
+          alt: item.login,
+        });
+        const p1 = createAndAppend('p', div, { id: 'p1' });
+        const p2 = createAndAppend('p', div, { id: 'p2' });
+        p1.textContent = item.login;
+        p2.innerHTML = item.contributions;
+        return item;
+      });
+    });
+  }
+
+  function startPage(data, listContainer, ul) {
+    createContributorsList(data.name, listContainer);
+    createInfoList(data, ul);
+  }
+
   function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root);
+    const section = createAndAppend('section', root);
+    createAndAppend('h1', header, { text: 'HYF Repositories' });
+    const ul = createAndAppend('ul', section, { id: 'list-container' });
+    const select = createAndAppend('select', header);
+    const listContainer = createAndAppend('ul', section, { id: 'Contributors_list' });
+    fetchJSON(url, (err, data) => {
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+      data
+        .sort((a, b) => a.name.localeCompare(b.name))
+        .forEach((item, index) => {
+          createAndAppend('option', select, { text: item.name, value: index });
+        });
+      startPage(data[0], listContainer, ul);
+
+      select.addEventListener('change', () => {
+        clearContainer(ul);
+        clearContainer(listContainer);
+        const repositoryData = data[select.value];
+        const repositoryName = data[select.value].name;
+        createContributorsList(repositoryName, listContainer);
+        createInfoList(repositoryData, ul);
```

Comment:
You already have a `startPage` function that does what lines 144-145 do. Why not reuse that function and make your code DRY?